rmldatatfbbox/rmldatatfbbox/utils/helpers.py

```diff
@@ -0,0 +1,115 @@
+            image_type = extension
+            ydim, xdim = tuple(cv2.imread(str(image_filepath.absolute())).shape[:2])
+            break
+
+    if image_filepath is None:
+        raise ValueError("Hmm, there doesn't seem to be a valid image filepath.")
+
+    with open(data_dir / f"meta_{image_id}.json", "r") as f:
+        meta = json.load(f)
+
+    label_boxes = []
+    for label, b in meta['bboxes'].items():
+        label_boxes.append({"label": label, "xmin": b['xmin'], "xmax": b['xmax'], "ymin": b['ymin'], "ymax": b['ymax']})
+        if label not in label_to_int_dict.keys():
+            label_to_int_dict[label] = len(label_to_int_dict)
+    
+    bboxes = {"image_id": image_id, "image_filepath": image_filepath, "image_type": image_type, "label_boxes": label_boxes, "xdim": xdim, "ydim": ydim}
+
+    return bboxes
+
+def write_label_map(dataset_name, out_dir, label_to_int_dict):
+        """Writes out the TensorFlow Object Detection Label Map
+        
+        Args:
+            dataset_name (str): the name of the dataset
+        """
+        dataset_path = out_dir / 'dataset' / dataset_name
+        label_map_filepath = dataset_path / 'label_map.pbtxt'
+        label_map = []
+        for label_name, label_int in label_to_int_dict.items():
+            label_info = "\n".join([
+                "item {", "  id: {id}".format(id=label_int),
+                "  name: '{name}'".format(name=label_name), "}"
+            ])
+            label_map.append(label_info)
+        with open(label_map_filepath, 'w') as outfile:
+            outfile.write("\n\n".join(label_map))
+
+def export_as_TFExample(object, label_to_int_dict):
+        """Converts LabeledImageMask object to tf_example
+        
+        Returns:
+            tf_example (tf.train.Example): TensorFlow specified training object.
+        """
+        path_to_image = Path(object["image_filepath"])
+
+        with tf.io.gfile.GFile(str(path_to_image), 'rb') as fid:
+            encoded_png = fid.read()
+
+        image_width  = object["xdim"]
+        image_height = object["ydim"]
+
+        filename = path_to_image.name.encode('utf8')
+        image_format = bytes(object["image_type"], encoding='utf-8')
+        xmins = []
+        xmaxs = []
+        ymins = []
+        ymaxs = []
+        classes_text = []
+        classes = []
+
+        for bounding_box in object["label_boxes"]:
+            xmins.append(bounding_box["xmin"] / image_width)
+            xmaxs.append(bounding_box["xmax"] / image_width)
+            ymins.append(bounding_box["ymin"] / image_height)
+            ymaxs.append(bounding_box["ymax"] / image_height)
+            classes_text.append(bounding_box["label"].encode('utf8'))
+            classes.append(label_to_int_dict[bounding_box["label"]])
+
+        tf_example = tf.train.Example(features=tf.train.Features(feature={
+            'image/height': dataset_util.int64_feature(image_height),
```

Comment:
We should break our dependency with object-detection. It makes absolutely no sense, because if you look at what `dataset_util` is actually doing, it's basically just aliases to easily accessible Tensorflow functions :facepalm:

Reply:
^ big bump, if we can just use TF that would be great. I always wondered why we needed OD for the datasets